### PR TITLE
Makefile should report non prod python versions

### LINKED ISSUE #304: Makefile should warn about mismatched python versions
Currently, we run Python 3.6 in production. However, that hasn't been captured in the project. It has been easy to use a different python version.

This really only makes a difference for dev & production -- CI doesn't matter as long as it supports the production version.

### PR DESCRIPTION
**Pull request by hwine**
Fixes #304

### CODE REVIEW

**Comment by ajvb** [comment]
@hwine I'd like to understand why developers should be warned about this? As well, it feels unneeded currently as we do not have any external contributors, even outside of the team, as of now.

Also, I'd like to bump frost and zap to using Python 3.8 in production here soon, going to open a PR soon for a Ubuntu 20.04 version of our base deployment container.

**Comment by ajvb** [comment]
Actually, thinking about it more, I'd rather we go for running tests against frost using python 3.6, 3.7, and 3.8.

**Comment by hwine** [comment]
@ajvb -- it's being proposed because it cost me time. I developed on 3.8 and that broke when it tried to run my code on 3.6.

My guess is it will be hard to _always_ have all 3 environments in sync (dev, CI, prod), this gives a way to notify folks of changes.

I'll be happy to see all those environment updated -- but is that any reason to not have a safety net?

I'll agree that supporting non-Mozilla users could lead to conflicts, but as you noted, that case doesn't yet exist.

**Comment by hwine** [comment]
> 
> 
> Actually, thinking about it more, I'd rather we go for running tests against frost using python 3.6, 3.7, and 3.8.

good, but different discussion -- opened #306 for it

**Comment by ajvb** [comment]
@hwine 

>@ajvb -- it's being proposed because it cost me time. I developed on 3.8 and that broke when it tried to run my code on 3.6.

Ok totally.

>good, but different discussion -- opened #306 for it

Is it a different discussion? I don't like the idea of having to keep the `PRODUCTION_PYTHON` variable synced up with what is being ran in production without automation around updating it.

If CI tests python 3.6+, then python 3.8 specific code would fail in CI before making it to production.

**Comment by hwine** [comment]
> If CI tests python 3.6+, then python 3.8 specific code would fail in CI before making it to production.

Agreed. But we didn't do that. We moved to 3.8 for CI without (imo) adequate notification that 3.6 is what matters for production.

To me, adding this warning (not error) is most visible when creating a venv, felt like the lowest friction way to address things. 

I _like_ the idea of getting on 3.8!!!  I wouldn't want to roll back to CI on 3.6 just to make things "consistent". The coordination of updating the three environments is already going to be challenging to find uninterrupted time for. Why create a "critical path" if we can avoid it?

> Is it a different discussion? I don't like the idea of having to keep the `PRODUCTION_PYTHON` variable synced up with what is being ran in production without automation around updating it.

I don't see us investing in automating upgrades across 3 environments. I see this action as a "checklist item" in the "run book" for doing the upgrade. Especially as more folks & teams get involved in using frost, the "explicit is better than implicit" guidance from `import this` seems worth it to me.

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,12 @@ AWS_PROFILE := default
 
 PYTEST_OPTS := ''
 
+PRODUCTION_PYTHON := 3.6
```
Comment (g-k) [inline Makefile:10]: nit: seems weird to mention production python in the OS repo. Should we just check for versions we don't test against in CI?

Comment (hwine) [reply Makefile:10]: agreed -- I'll switch to PYTHON_MIN_VERSION

**Comment by g-k** [comment]
Thanks Hal!

**Pull request merged**